server/graphql/queries.js

```diff
@@ -640,6 +650,44 @@ const queries = {
         }
       });
     }
+  },
+
+  /*
+   * Given a search term, return a list of related Collectives
+   */
+  search: {
+    type: new GraphQLList(CollectiveSearchType),
```

Comment:
why not returning a list of `CollectiveInterfaceType`?
could use algolia to return the `ids` and then fetch the corresponding collectives. That way we keep the number of GraphQL types small.